ngraph/frontend/generic/include/frontend_manager/frontend_manager.hpp

```diff
@@ -0,0 +1,454 @@
+            typedef std::shared_ptr<Place> Ptr;
+
+            virtual ~Place() = default;
+
+            /// \brief All associated names (synonyms) that identify this place in the graph in a
+            /// framework specific way
+            ///
+            /// \return A vector of strings each representing a name that identifies this place in
+            /// the graph. Can be empty if there are no names associated with this place or name
+            /// cannot be attached.
+            virtual std::vector<std::string> get_names() const;
+
+            /// \brief Returns references to all operation nodes that consume data from this place
+            /// \note It can be called for any kind of graph place searching for the first consuming
+            /// operations.
+            ///
+            /// \param outputPortIndex If place is an operational node it specifies which output
+            /// port should be considered
+            ///
+            /// \return A vector with all operation node references that consumes data from this
+            /// place
+            virtual std::vector<Ptr> get_consuming_operations(int outputPortIndex = -1) const;
+
+            /// \brief Returns a tensor place that gets data from this place; applicable for
+            /// operations, output ports and output edges
+            ///
+            /// \param outputPortIndex Output port index if the current place is an operation node
+            /// and has multiple output ports
+            ///
+            /// \return A tensor place which hold the resulting value for this place
+            virtual Ptr get_target_tensor(int outputPortIndex = -1) const;
+
+            /// \brief Returns a tensor place that supplies data for this place; applicable for
+            /// operations, input ports and input edges
+            ///
+            /// \param inputPortIndex Input port index for operational nodes
+            /// \return A tensor place which supplies data for this place
+            virtual Ptr get_source_tensor(int inputPortIndex = -1) const;
+
+            /// \brief Get an operation node place that immediately produces data for this place
+            ///
+            /// \param inputPortIndex If a given place is itself an operation node, this specifies a
+            /// port index
+            ///
+            /// \return An operation place that produces data for this place
+            virtual Ptr get_producing_operation(int inputPortIndex = -1) const;
+
+            /// Returns a port that produces data for this place
+            virtual Ptr get_producing_port() const;
+
+            /// For operation node returns reference to an input port with specified index
+            virtual Ptr get_input_port(int inputPortIndex = -1) const;
+
+            /// For operation node returns reference to an input port with specified name and index
+            virtual Ptr get_input_port(const std::string& inputName, int inputPortIndex = -1) const;
```

Comment:
We don't have a description for parameters and the same question for -1 value.

Reply:
inputName is a name of the port group, in each group there may be multiple ports that фку distinguished with `inputPortIndex`. The same answer for -1. And this case illustrates it is better to keep capability to access port by default index if there is only a single output, because in this case, name provides 'non default' port specification.